File: BaekJoon/28014_첨탑밀어서부수기_Greedy.cpp
#include <iostream>

using namespace std;

// 첨탑밀어서부수기 28014 Greedy
// O( N )


int main(void)
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;
    cin >> n;

    int height, prevHeight; // 현재 높이, 이전 높이
    cin >> prevHeight;

    int cnt{ 1 };   // 민 횟수
    for (int i = 1; i < n; ++i)
    {
        cin >> height;
        // 현재 높이가 이전 높이 보다 크거나 같다면 밀어야 함
        if (height >= prevHeight)
        {
            ++cnt;
        }
        prevHeight = height;    // 이전 높이 갱신
    }

    cout << cnt;


    return 0;
}

File: BaekJoon/Java/28014_첨탑밀어서부수기_Greedy.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

// 첨탑밀어서부수기 28014 Greedy
// O( N )


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null;

        int n = Integer.parseInt(br.readLine());

        st = new StringTokenizer(br.readLine());

        int prevHeight = Integer.parseInt(st.nextToken());
        int cnt = 1;
        for (int i = 1; i < n; ++i) {
            int curHeight = Integer.parseInt(st.nextToken());
            if (curHeight >= prevHeight) {
                ++cnt;
            }
            prevHeight = curHeight;
        }

        System.out.println(cnt);
    }
}
